wallet/wallet-node-client/src/node_traits.rs

```diff
@@ -38,6 +38,11 @@ pub trait NodeInterface {
         first_block: Id<GenBlock>,
         second_block: Id<GenBlock>,
     ) -> Result<Option<(Id<GenBlock>, BlockHeight)>, Self::Error>;
+    async fn generate_block(
+        &self,
+        reward_destination_hex: String,
+        transactions_hex: Option<Vec<String>>,
```

Comment:
One thing that isn't done correctly in this trait and should be fixed later (or feel free to fix it now if you wish), is the assumption that everything has to be serialized. The serialization should be implemented in the RPC implementation, not in this trait. This is because if we use handles directly, then we won't have to serialize anything.

Reply:
OK, sure, but let's do this in other PR.

Reply:
Should be fixed here - https://github.com/mintlayer/mintlayer-core/pull/898